### DISCUSSION TITLE
REQ: race hud ui

### DISCUSSION THREAD

**Discussion by meag** [ezQuake/ezquake-source#498] [Ideas]
**Describe the solution you'd like**
fourier:
Is is possible to make ezquake ui for race mode? i.e. button to turn on the race mode when on a server, show table with personal bests/global best races for a map, tutorial mode, personal progression graph with date/times of persobal bests for a day and number of attempts etc
I guess collecting stats should be done on a server side, but client could have a ui to present it nicely

would require ktx support

**Comment by ShondaBecker** [comment]
This is a really interesting idea and it makes a lot of sense for race mode specifically. Having a dedicated UI toggle would already be a big quality-of-life improvement, especially for players jumping between normal play and racing. The idea of showing personal bests alongside global records would add a lot of motivation, and a tutorial mode would help newer players understand routes and mechanics faster. A progression graph with dates, attempts, and improvements over time would be great for tracking growth and staying engaged. Fastpay delivers quick https://fastpaycasinoau.com/ transactions and smooth, hassle-free gameplay. You’re right that most of the stat collection would need to be server-side, but a clean client-side UI could make all of that data feel much more accessible and rewarding.

**Comment by JesseReed123** [comment]
It would be really useful to have an EZQuake UI specifically for race mode. Ideally, it could include a button to enable race mode on a server, a table showing personal bests and global best times for each map, a tutorial mode, and a personal progression graph displaying dates and times of personal bests along with the number of attempts per day. While the stats https://roobet-in.com/ collection would likely need to be handled server-side, the client could present all this information in a clear, user-friendly interface, making it much easier to track and improve performance.
